api/src/main/java/com/messagebird/MessageBirdClient.java

```diff
@@ -1189,6 +1217,83 @@ public RecordingResponse viewRecording(String callID, String legId, String recor
         return messageBirdService.requestByID(url, callID, params, RecordingResponse.class);
     }
 
+    /**
+     * Downloads the record in .wav format by using callId, legId and recordId and stores to basePath
+     *
+     * @param callID Voice call ID
+     * @param legId Leg ID
+     * @param recordingId Recording ID
+     * @param basePath store location. It should be directory
+     * @return the path that file is stored
+     * @throws NotFoundException
+     * @throws GeneralException
+     * @throws UnauthorizedException
+     */
+    public String downloadRecording(String callID, String legId, String recordingId, String basePath) throws NotFoundException, GeneralException, UnauthorizedException {
+
+        if (callID == null) {
+            throw new IllegalArgumentException("Voice call ID must be specified.");
+        }
+
+        if (legId == null) {
+            throw new IllegalArgumentException("Leg ID must be specified.");
+        }
+
+        if (recordingId == null) {
+            throw new IllegalArgumentException("Recording ID must be specified.");
+        }
+
```

Comment:
There is no basePath null check but let's use basePath param optional. By default, download endpoint can let download the recording in downloads directly. If customer wants to specify path then they can give basePath variable. What do you think? We can make this change because there is no specific constraint on developer doc.